src/VisualStudio/CSharp/Impl/LanguageService/CSharpHelpContextService.cs

```diff
@@ -451,10 +451,19 @@ private static bool TryGetTextForKeyword(SyntaxToken token, [NotNullWhen(true)]
                 }
             }
 
-            if (token.IsKind(SyntaxKind.DefaultKeyword) && token.Parent is DefaultSwitchLabelSyntax)
+            if (token.IsKind(SyntaxKind.DefaultKeyword))
             {
-                text = Keyword("defaultcase");
-                return true;
+                if (token.Parent is DefaultSwitchLabelSyntax)
+                {
+                    text = Keyword("defaultcase");
+                    return true;
+                }
+
+                if (token.Parent is DefaultConstraintSyntax)
+                {
+                    text = Keyword("defaultconstraint");
+                    return true;
+                }
```

Comment:
Consider handling `goto default` (not sure which doc page is suitable).

Also `#line default`

Reply:
Thanks for the suggestions. Added

I'll let @BillWagner decide, but for `goto default` we could always go to the [page](https://learn.microsoft.com/en-US/dotnet/csharp/language-reference/statements/selection-statements) for `switch` and for `#line default` we could go to the [section](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives#error-and-warning-information) for #line preprocessing directive.

Reply:
> we could go to the [section](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/preprocessor-directives#error-and-warning-information)

We can't go to a specific section, unfortunately.

But I'll update the page on dotnet/docs to have the new F1 keyword.